File: algorithms_in_matlab/week_9/ex8/originalGaussian.m
function p = originalGaussian(X, mu, Sigma2)
%
%p(x)=p(x1;mu1,sig²1) * p(x2;mu2,sig²2) ... p(xn;mun,sig²n)
%
p =  (1./(sqrt(2 .* pi .* Sigma2))) .* exp(- ((X - mu) .^ 2) ./ (2 * Sigma2));
p = prod(p, 2); %  products of each row
end

File: algorithms_in_python/week_9/ex8/originalGaussian.py
import math

import numpy as np


def originalGaussian(X, mu, Sigma2):
    exp_data = ((X - mu) ** 2) / (2 * Sigma2)
    lef_data = (1.0 / (np.sqrt(2 * math.pi * Sigma2)))
    p = lef_data * np.exp(- exp_data)
    p = np.prod(p, 1)  # products of each row
    return p
